### PR TITLE
[#IOPID-1315] add opex dashboard for external openapi

### PR DESCRIPTION
**Pull request by arcogabbo**
#### List of Changes

* [[#IOPID-1315] add github actions](https://github.com/pagopa/io-web-profile-backend/commit/d5d63bf9469291ca3fc8952c1331b85814fdc15e)
* [[#IOPID-1315] fix: remove env directory from gitignore to support ter…](https://github.com/pagopa/io-web-profile-backend/commit/eafde4d62ad092f52e26bb4ffc24a941759401e3)
* [[#IOPID-1315] add external.yaml opex dashboards](https://github.com/pagopa/io-web-profile-backend/commit/e2785322ec3dc2a8d6679cc64285f963b2561cb2)

#### Motivation and Context

add opex dashboards for `external.yaml` openapi

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .identity/.gitignore
Comment (krusty93) [inline .identity/.gitignore:1]: I wouldn't add another .gitignore, I'd add instead TF related things to the main one

File: .github/workflows/pr_opex_external.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: PR - Opex Dashboard External
+
+on:
+  pull_request:
+    types:
+      - opened
+      - edited
+      - synchronize
+      - reopened
```
Comment (krusty93) [inline .github/workflows/pr_opex_external.yml:9]: Add `ready_for_review`

File: .identity/00_main.tf [hunk 2]
```diff
@@ -0,0 +1,32 @@
+terraform {
+  required_version = ">=1.3.0"
```
Comment (krusty93) [inline .identity/00_main.tf]: 1.6.0

File: .identity/00_main.tf.ci
Comment (krusty93) [inline .identity/00_main.tf.ci:1]: What's this file? Can be deleted

File: .identity/00_variables.tf [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  }))
+  description = "GitHub Organization, repository name and scope permissions"
+}
+
+variable "environment_ci_roles" {
+  type = object({
+    subscription    = list(string)
+    resource_groups = map(list(string))
+  })
+  description = "GitHub Continous Delivery roles"
```
Comment (krusty93) [inline .identity/00_variables.tf]: `Integration`

File: .identity/02_application_master.tf
Comment (krusty93) [inline .identity/02_application_master.tf:1]: Delete this file, it's an old approach

File: .identity/02_application_master_auth.tf
Comment (krusty93) [inline .identity/02_application_master_auth.tf:1]: Delete this file, it's an old approach

File: .identity/02_application_pullrequest.tf
Comment (krusty93) [inline .identity/02_application_pullrequest.tf:1]: Delete this file, it's an old approach

File: .identity/02_application_pullrequest_auth.tf
Comment (krusty93) [inline .identity/02_application_pullrequest_auth.tf:1]: Delete this file, it's an old approach

File: .identity/03_managed_identity_cd.tf [hunk 4]
```diff
@@ -0,0 +1,18 @@
+module "identity_cd" {
+  source = "github.com/pagopa/terraform-azurerm-v3//github_federated_identity?ref=v7.34.0"
```
Comment (krusty93) [inline .identity/03_managed_identity_cd.tf]: Get the latest version

File: .identity/03_managed_identity_ci.tf [hunk 5]
```diff
@@ -0,0 +1,18 @@
+module "identity_ci" {
+  source = "github.com/pagopa/terraform-azurerm-v3//github_federated_identity?ref=v7.34.0"
```
Comment (krusty93) [inline .identity/03_managed_identity_ci.tf]: Get latest version

File: .identity/03_managed_identity_ci.tf [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+  prefix    = var.prefix
+  env_short = var.env_short
+  domain    = var.domain
+
+  identity_role = "ci"
+
+  github_federations = var.ci_github_federations
+
+  cd_rbac_roles = {
```
Comment (krusty93) [inline .identity/03_managed_identity_ci.tf]: `ci_rbac_roles`

File: .identity/env/prod/backend.tfvars [hunk 7]
```diff
@@ -0,0 +1,4 @@
+resource_group_name  = "terraform-state-rg"
+storage_account_name = "tfappprodio"
+container_name       = "terraform-state"
+key                  = "io-web-profile-backend.tfstate"
```
Comment (krusty93) [inline .identity/env/prod/backend.tfvars]: `io-web-profile-backend.identity.tfstate`

File: .opex/external/env/prod/backend.tfvars [hunk 8]
```diff
@@ -0,0 +1,4 @@
+resource_group_name  = "terraform-state-rg"
+storage_account_name = "tfappprodio"
+container_name       = "terraform-state"
+key                  = "opex.io-web-profile-backend.external.tfstate"
```
Comment (krusty93) [inline .opex/external/env/prod/backend.tfvars:4]: `io-web-profile-backend.opex.tfstate`?

File: .identity/env/prod/terraform.tfvars [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+      "Reader"
+    ]
+  }
+}
+
+environment_cd_roles = {
+  subscription = ["Reader"]
+  resource_groups = {
+    "terraform-state-rg" = [
+      "Storage Blob Data Contributor"
```
Comment (krusty93) [inline .identity/env/prod/terraform.tfvars]: Add `Reader and Data Access`

File: .identity/00_variables.tf [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+variable "ci_github_federations" {
+  type = list(object({
+    repository        = string
+    credentials_scope = optional(string, "environment")
+    subject           = string
+  }))
+  description = "GitHub Organization, repository name and scope permissions"
+}
+
+variable "environment_ci_roles" {
```
Comment (krusty93) [inline .identity/00_variables.tf]: Please rename to `opex_environment_ci_roles`

File: .identity/00_variables.tf [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+variable "cd_github_federations" {
+  type = list(object({
+    repository        = string
+    credentials_scope = optional(string, "environment")
+    subject           = string
+  }))
+  description = "GitHub Organization, repository name and scope permissions"
+}
+
+variable "environment_cd_roles" {
```
Comment (krusty93) [inline .identity/00_variables.tf]: Please rename to `opex_environment_cd_roles`

File: .identity/03_managed_identity_ci.tf [hunk 12]
```diff
@@ -0,0 +1,18 @@
+module "identity_ci" {
+  source = "github.com/pagopa/terraform-azurerm-v3//github_federated_identity?ref=v7.34.0"
+
+  prefix    = var.prefix
+  env_short = var.env_short
+  domain    = var.domain
+
+  identity_role = "ci"
+
+  github_federations = var.ci_github_federations
```
Comment (krusty93) [inline .identity/03_managed_identity_ci.tf]: If you wish, you can rid of this variable and use something like:

```hcl
github_federations = [
    {
      repository = "io-web-profile-backend"
      subject    = github_repository_environment.prod_ci.environment
    }
  ]
  ```

File: .identity/04_github_environment.tf [hunk 13]
```diff
@@ -0,0 +1,32 @@
+resource "github_repository_environment" "prod_ci" {
```
Comment (krusty93) [inline .identity/04_github_environment.tf]: I'd prefer to add an `opex` suffix somewhere, like [this](https://github.com/pagopa/io-sign/blob/main/.identity/github_environment_opex.tf)

File: .identity/01_data.tf
Comment (krusty93) [inline .identity/01_data.tf:1]: Delete this file, it's an old approach

**Review by krusty93 — CHANGES_REQUESTED** [review]

File: .identity/03_managed_identity_ci.tf (hunk 6 above)
Comment (arcogabbo) [reply .identity/03_managed_identity_ci.tf]: fixed in https://github.com/pagopa/io-web-profile-backend/pull/32/commits/4f8a7668f8d9035acf20e7ef30df8b6cd26ccc3d

File: .identity/env/prod/terraform.tfvars (hunk 9 above)
Comment (arcogabbo) [reply .identity/env/prod/terraform.tfvars]: done in https://github.com/pagopa/io-web-profile-backend/pull/32/commits/4f8a7668f8d9035acf20e7ef30df8b6cd26ccc3d

File: .identity/.gitignore
Comment (arcogabbo) [reply .identity/.gitignore:1]: done in [7020a5d](https://github.com/pagopa/io-web-profile-backend/pull/32/commits/7020a5d21d9f1bee6185a78d5341d01d9227fd68)

File: .identity/00_main.tf.ci
Comment (arcogabbo) [reply .identity/00_main.tf.ci:1]: done in [3c1213e](https://github.com/pagopa/io-web-profile-backend/pull/32/commits/3c1213ed283ebc463136e19f8fe7fd2d1fa9f448)

File: .identity/02_application_master.tf
Comment (arcogabbo) [reply .identity/02_application_master.tf:1]: done in [3c1213e](https://github.com/pagopa/io-web-profile-backend/pull/32/commits/3c1213ed283ebc463136e19f8fe7fd2d1fa9f448)

File: .identity/02_application_master_auth.tf
Comment (arcogabbo) [reply .identity/02_application_master_auth.tf:1]: done in [3c1213e](https://github.com/pagopa/io-web-profile-backend/pull/32/commits/3c1213ed283ebc463136e19f8fe7fd2d1fa9f448)

File: .identity/02_application_pullrequest.tf
Comment (arcogabbo) [reply .identity/02_application_pullrequest.tf:1]: done in [3c1213e](https://github.com/pagopa/io-web-profile-backend/pull/32/commits/3c1213ed283ebc463136e19f8fe7fd2d1fa9f448)

File: .identity/02_application_pullrequest_auth.tf
Comment (arcogabbo) [reply .identity/02_application_pullrequest_auth.tf:1]: done in [3c1213e](https://github.com/pagopa/io-web-profile-backend/pull/32/commits/3c1213ed283ebc463136e19f8fe7fd2d1fa9f448)

File: .identity/01_data.tf
Comment (arcogabbo) [reply .identity/01_data.tf:1]: done in [3c1213e](https://github.com/pagopa/io-web-profile-backend/pull/32/commits/3c1213ed283ebc463136e19f8fe7fd2d1fa9f448)

File: .opex/external/env/prod/backend.tfvars (hunk 8 above)
Comment (arcogabbo) [reply .opex/external/env/prod/backend.tfvars:4]: `external` is the name of the openapi. the goal is to add more apis in the future. `opex` is too generic in this case. would you suggest another name?

File: .github/workflows/pr_opex_external.yml (hunk 1 above)
Comment (arcogabbo) [reply .github/workflows/pr_opex_external.yml:9]: done in [42a1bd0](https://github.com/pagopa/io-web-profile-backend/pull/32/commits/42a1bd07a2f1b3618634d703ecbf89c979516510)

File: .identity/00_main.tf (hunk 2 above)
Comment (arcogabbo) [reply .identity/00_main.tf]: done in [42a1bd0](https://github.com/pagopa/io-web-profile-backend/pull/32/commits/42a1bd07a2f1b3618634d703ecbf89c979516510)

File: .identity/00_variables.tf (hunk 3 above)
Comment (arcogabbo) [reply .identity/00_variables.tf]: fixed in [42a1bd0](https://github.com/pagopa/io-web-profile-backend/pull/32/commits/42a1bd07a2f1b3618634d703ecbf89c979516510)

File: .identity/03_managed_identity_cd.tf (hunk 4 above)
Comment (arcogabbo) [reply .identity/03_managed_identity_cd.tf]: done in [42a1bd0](https://github.com/pagopa/io-web-profile-backend/pull/32/commits/42a1bd07a2f1b3618634d703ecbf89c979516510)

File: .identity/03_managed_identity_ci.tf (hunk 5 above)
Comment (arcogabbo) [reply .identity/03_managed_identity_ci.tf]: done in [42a1bd0](https://github.com/pagopa/io-web-profile-backend/pull/32/commits/42a1bd07a2f1b3618634d703ecbf89c979516510)

File: .identity/env/prod/backend.tfvars (hunk 7 above)
Comment (arcogabbo) [reply .identity/env/prod/backend.tfvars]: done in [42a1bd0](https://github.com/pagopa/io-web-profile-backend/pull/32/commits/42a1bd07a2f1b3618634d703ecbf89c979516510)

File: .identity/00_variables.tf (hunk 10 above)
Comment (arcogabbo) [reply .identity/00_variables.tf]: done in [42a1bd0](https://github.com/pagopa/io-web-profile-backend/pull/32/commits/42a1bd07a2f1b3618634d703ecbf89c979516510)

File: .identity/00_variables.tf (hunk 11 above)
Comment (arcogabbo) [reply .identity/00_variables.tf]: done in [42a1bd0](https://github.com/pagopa/io-web-profile-backend/pull/32/commits/42a1bd07a2f1b3618634d703ecbf89c979516510)

File: .identity/03_managed_identity_ci.tf (hunk 12 above)
Comment (arcogabbo) [reply .identity/03_managed_identity_ci.tf]: thank you, done in [42a1bd0](https://github.com/pagopa/io-web-profile-backend/pull/32/commits/42a1bd07a2f1b3618634d703ecbf89c979516510)

File: .identity/04_github_environment.tf (hunk 13 above)
Comment (arcogabbo) [reply .identity/04_github_environment.tf]: done in [42a1bd0](https://github.com/pagopa/io-web-profile-backend/pull/32/commits/42a1bd07a2f1b3618634d703ecbf89c979516510)

File: .identity/00_variables.tf [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+
+variable "env_short" {
+  type = string
+}
+
+variable "domain" {
+  type = string
+}
+
+variable "ci_github_federations" {
```
Comment (krusty93) [inline .identity/00_variables.tf]: Not used anymore, can be deleted

File: .identity/00_variables.tf [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+
+variable "opex_environment_ci_roles" {
+  type = object({
+    subscription    = list(string)
+    resource_groups = map(list(string))
+  })
+  description = "GitHub Continous Integration roles"
+}
+
+variable "cd_github_federations" {
```
Comment (krusty93) [inline .identity/00_variables.tf]: Not used anymore, can be deleted

File: .identity/04_github_environment.tf [hunk 16]
```diff
@@ -0,0 +1,32 @@
+resource "github_repository_environment" "prod_opex_ci" {
+  environment = "${var.env}-ci"
```
Comment (krusty93) [inline .identity/04_github_environment.tf]: Sorry I didn't notice it before, but I'd like to change the name `opex-${var.env}-ci` as done in `io-sign`

File: .identity/04_github_environment.tf [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+  repository  = local.github.repository
+}
+
+resource "github_repository_environment" "prod_opex_cd" {
+  environment = "${var.env}-cd"
```
Comment (krusty93) [inline .identity/04_github_environment.tf]: Sorry I didn't notice it before, but I'd like to change the name `opex-${var.env}-cd` as done in `io-sign`

File: .identity/env/prod/terraform.tfvars [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+
+tags = {
+  CreatedBy   = "Terraform"
+  Environment = "Prod"
+  Owner       = "io"
+  Source      = "https://github.com/pagopa/io-web-profile-backend"
+  CostCenter  = "TS310 - PAGAMENTI & SERVIZI"
+}
+
+ci_github_federations = [
```
Comment (krusty93) [inline .identity/env/prod/terraform.tfvars]: Can be removed (see other comments in variables)

File: .opex/external/env/prod/backend.tfvars (hunk 8 above)
Comment (krusty93) [reply .opex/external/env/prod/backend.tfvars:4]: Looking at what `io-sign` does, I'd say `"opex.io-web-profile-backend.opex.external.tfstate"` to still include it under the `opex` umbrella. What do you think?

Comment (arcogabbo) [reply .opex/external/env/prod/backend.tfvars:4]: with the one you suggested `opex` keyword is mentioned 2 times(as a prefix and before `.external`)

File: .identity/env/prod/terraform.tfvars (hunk 18 above)
Comment (arcogabbo) [reply .identity/env/prod/terraform.tfvars]: done in [14ee036](https://github.com/pagopa/io-web-profile-backend/pull/32/commits/14ee0365658a58dfc25fe0c9fbb78d25f4fb2503)

File: .identity/04_github_environment.tf (hunk 17 above)
Comment (arcogabbo) [reply .identity/04_github_environment.tf]: done in [14ee036](https://github.com/pagopa/io-web-profile-backend/pull/32/commits/14ee0365658a58dfc25fe0c9fbb78d25f4fb2503)

File: .identity/04_github_environment.tf (hunk 16 above)
Comment (arcogabbo) [reply .identity/04_github_environment.tf]: done in [14ee036](https://github.com/pagopa/io-web-profile-backend/pull/32/commits/14ee0365658a58dfc25fe0c9fbb78d25f4fb2503)

File: .identity/00_variables.tf (hunk 15 above)
Comment (arcogabbo) [reply .identity/00_variables.tf]: done in [14ee036](https://github.com/pagopa/io-web-profile-backend/pull/32/commits/14ee0365658a58dfc25fe0c9fbb78d25f4fb2503)

File: .identity/00_variables.tf (hunk 14 above)
Comment (arcogabbo) [reply .identity/00_variables.tf]: done in [14ee036](https://github.com/pagopa/io-web-profile-backend/pull/32/commits/14ee0365658a58dfc25fe0c9fbb78d25f4fb2503)

File: .opex/external/env/prod/backend.tfvars (hunk 8 above)
Comment (krusty93) [reply .opex/external/env/prod/backend.tfvars:4]: Sorry - a typo. I meant `opex.io-web-profile-backend.external.tfstate`

Comment (arcogabbo) [reply .opex/external/env/prod/backend.tfvars:4]: it's already as you mentioned

**Review by gquadrati — APPROVED** [review]
LGTM 🚀

**Review by krusty93 — APPROVED** [review]

**Pull request merged**